middleware/protect.js

```diff
@@ -17,11 +17,11 @@
 
 const UUID = require('./../uuid');
 
-function forceLogin (keycloak, request, response) {
-  let host = request.hostname;
+function forceLogin (keycloak, request, response, next) {
   let headerHost = request.headers.host.split(':');
+  let host = request.hostname || headerHost[0];
   let port = headerHost[1] || '';
-  let protocol = request.protocol;
+  let protocol = (request.protocol !== undefined) ? request.protocol : request.isSecure() ? 'https' : 'http';
```

Comment:
does `request.isSecure` exist on node http or express?
what about `request.connection.secure`?
what if the application is behind proxy?

Reply:
that was my main headache on the patch.. seems like `protocol` comes on `express`, therefore I left it to apply as it was before the patch. 

`isSecure()` is wrapping the `encrypted`, based on this [doc](http://restify.com/docs/request-api/#issecure), 
should cover the cases behind proxy for `restify` too, 

more or less I used that toggle in that line to somehow tell us if it's `restify` or not, I couldn't find a prettier way to do that..